core/src/subgraph/runner.rs

```diff
@@ -185,8 +185,20 @@ where
 
                 let (block, cursor) = match event {
                     Some(Ok(BlockStreamEvent::ProcessBlock(block, cursor))) => (block, cursor),
-                    Some(Ok(BlockStreamEvent::Revert(subgraph_ptr, revert_to_ptr, cursor))) => {
-                        info!(&logger, "Reverting block to get back to main chain"; "current" => &subgraph_ptr, "revert_to" => &revert_to_ptr);
+                    // In the case of the PollingBlockStream, the stream_ptr can be ahead of the
+                    // subgraph's deployment head (database), because this component is pipelined.
+                    Some(Ok(BlockStreamEvent::Revert(stream_ptr, revert_to_ptr, cursor))) => {
```

Comment:
Should we remove the `stream_ptr` entirely then, it's not used anymore expect for logging purposes, so maybe it's not even worth it have it returned anymore?

Reply:
I was wondering the same, but I didn't remove because I wasn't sure about the firehose part.

Reply:
Firehose could be pipelined itself (when you say `pipeline` you are referring to wrapping through the`BufferedBlockStream` right), so it shouldn't matter here if we are through Firehose or not.

The only difference is that FirehoseBlockStream can revert more than 1 block, just as an info.

And as a side note, and I think that when we have "filtering/trimming" in place to have Firehose, `BufferedBlockStream` default value for Firehose should be at least 2, so we do the "decoding" work in parallel to always have block ready in the pipeline (right now each block are "full", so it's probably a bad idea to have each subgraph maybe having 2 blocks ready, will see in the integration testing in the upcoming weeks :D)

Reply:
I agree we can remove the first parameter of `BlockStreamEvent::Revert`.

Reply:
I just removed the field, thanks for pointing this out @maoueh ❤️